packages/eslint-plugin/src/rules/consistent-type-definitions.ts

```diff
@@ -68,15 +71,27 @@ export default createRule({
                 );
               }
 
-              const afterToken = context.sourceCode.getTokenAfter(
-                node.typeAnnotation,
-              );
-              if (
-                afterToken &&
-                afterToken.type === AST_TOKEN_TYPES.Punctuator &&
-                afterToken.value === ';'
-              ) {
-                fixes.push(fixer.remove(afterToken));
+              // remove all closing parens, and the semicolon if present.
+              let rangeStart = node.typeAnnotation.range[1];
+              while (true) {
+                const afterToken =
+                  context.sourceCode.getTokenByRangeStart(rangeStart);
+
+                if (afterToken != null) {
+                  if (afterToken.value === ')') {
+                    fixes.push(fixer.remove(afterToken));
+                    rangeStart = afterToken.range[1];
+                    continue;
+                  } else if (
+                    afterToken.type === AST_TOKEN_TYPES.Punctuator &&
+                    afterToken.value === ';'
+                  ) {
+                    fixes.push(fixer.remove(afterToken));
+                    break;
+                  }
+                }
+
+                break;
               }
```

Comment:
The better way to do this might be to do (psuedocode):
1) `getTokenBefore(typeAnnotation, '=')`
2) `getTokenAfter(typeAnnotation, ';')`
3) replace range from (1) to (2) with `typEAnnotation`

Reply:
^Agreed; this feels a bit hacky

Reply:
yeah, this was a good call. Result is much cleaner now 👍